### PR TITLE
Dynamodb source fix

### PR DESCRIPTION
**Pull request by OnkarVO7**
### Describe your changes :

Dynamodb source fix

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@harshach @ayush-shah @pmbrull

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/dynamodb.py [hunk 1]
```diff
@@ -50,13 +44,20 @@ def __init__(
...
+        ).get_resource("dynamodb")
 
     @classmethod
     def create(cls, config_dict, metadata_config: OpenMetadataServerConfig):
-        config = DynamoDBSourceConfig.parse_obj(config_dict)
+        config: WorkflowSource = WorkflowSource.parse_obj(config_dict)
+        connection: DynamoDBConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, DynamoDBConnection):
+            raise InvalidSourceException(
+                f"Expected SQLiteConnection, but got {connection}"
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/dynamodb.py:59]: Fix this name here pls for DynamoDB

File: ingestion/src/metadata/ingestion/source/dynamodb.py [hunk 2]
```diff
@@ -77,18 +78,22 @@ def ingest_tables(self, next_tables_token=None) -> Iterable[OMetaDatabaseAndTabl
         tables = list(self.dynamodb.tables.all())
         for table in tables:
             try:
-                if not self.config.table_filter_pattern.included(table.name):
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/dynamodb.py:80]: The filtering logic is now centralised https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/utils/filters.py let's see if we can use that or update it if required 🙏

File: ingestion/src/metadata/ingestion/source/dynamodb.py [hunk 3]
```diff
@@ -77,18 +78,22 @@ def ingest_tables(self, next_tables_token=None) -> Iterable[OMetaDatabaseAndTabl
...
                     continue
                 database_entity = Database(
-                    name=self.config.db_name,
+                    id=uuid.uuid4(),
+                    name="default",
                     service=EntityReference(id=self.service.id, type="databaseService"),
                 )
 
-                fqn = f"{self.config.service_name}{FQDN_SEPARATOR}{database_entity.name}{FQDN_SEPARATOR}{table}"
+                fqn = f"{self.config.serviceName}{FQDN_SEPARATOR}{database_entity.name}{FQDN_SEPARATOR}{table}"
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/dynamodb.py:96]: Do not inform the FQN. It is optional in `Table` and the backend will regenerate it

**Review by pmbrull — CHANGES_REQUESTED** [review]
Thanks for the PR @OnkarVO7, a few fast comments pls!

File: ingestion/src/metadata/ingestion/source/dynamodb.py (hunk 1 above)
Comment (OnkarVO7) [reply ingestion/src/metadata/ingestion/source/dynamodb.py:59]: okay

File: ingestion/src/metadata/ingestion/source/dynamodb.py (hunk 2 above)
Comment (OnkarVO7) [reply ingestion/src/metadata/ingestion/source/dynamodb.py:80]: yes I'll check it

**Review by pmbrull — APPROVED** [review]

**Pull request merged**